rollout/trafficrouting/istio/istio.go

```diff
@@ -1442,7 +1442,7 @@ func getOrderedVirtualServiceRoutes(httpRouteI []interface{}, managedRoutes []v1
 
 			// No need to check if exist because the empty string returned on cast failure is good for this check
 			name, _ := r["name"].(string)
-			if name == routeTyped.Name {
+			if name == routeTyped["name"] || (name == "" && routeTyped["name"] == nil) {
```

Comment:
same issue with type safety mentioned before

Reply:
Changed this logic a bit to do a type cast and use the failure case for the special case of no named route.